client/my-sites/upgrades/domain-search/domain-search.jsx

```diff
@@ -1,95 +1,85 @@
+import QueryProductsList from 'components/data/query-products-list';
 
-var DomainSearch = React.createClass( {
-	mixins: [ observe( 'productsList', 'sites' ), analyticsMixin( 'registerDomain' ) ],
+const analytics = analyticsMixin( 'registerDomain' );
 
-	propTypes: {
-		sites: React.PropTypes.object.isRequired,
+class DomainSearch extends Component {
+	static propTypes = {
+		selectedSite: React.PropTypes.object.isRequired,
+		sitePlans: React.PropTypes.object.isRequired,
 		productsList: React.PropTypes.object.isRequired,
 		basePath: React.PropTypes.string.isRequired,
 		context: React.PropTypes.object.isRequired,
 		domainsWithPlansOnly: React.PropTypes.bool.isRequired
-	},
-
-	getInitialState: function() {
-		return { domainRegistrationAvailable: true };
-	},
+	};
+
+	constructor() {
+		super();
+		this.handleDomainsAvailabilityChange = this.handleDomainsAvailabilityChange.bind( this );
+		this.handleAddRemoveDomain = this.handleAddRemoveDomain.bind( this );
+		this.handleAddMapping = this.handleAddMapping.bind( this );
+		this.state = { domainRegistrationAvailable: true };
+	}
 
-	componentWillMount: function() {
+	componentDidMount() {
 		this.checkSiteIsUpgradeable();
-	},
-
-	componentDidMount: function() {
-		this.props.sites.on( 'change', this.checkSiteIsUpgradeable );
-		this.props.fetchSitePlans( this.props.sitePlans, this.props.sites.getSelectedSite() );
-
-		this.previousSelectedSite = this.props.sites.getSelectedSite();
-	},
-
-	componentWillReceiveProps: function() {
-		var selectedSite = this.props.sites.getSelectedSite();
-		if ( this.previousSelectedSite !== selectedSite ) {
-			this.props.fetchSitePlans( this.props.sitePlans, selectedSite );
-			this.previousSelectedSite = selectedSite;
-		}
-	},
-
-	componentWillUnmount: function() {
-		this.props.sites.off( 'change', this.checkSiteIsUpgradeable );
-	},
+	}
 
-	checkSiteIsUpgradeable: function() {
-		var selectedSite = this.props.sites.getSelectedSite();
+	componentWillReceiveProps() {
+		this.checkSiteIsUpgradeable();
+	}
 
-		if ( selectedSite && ! selectedSite.isUpgradeable() ) {
+	checkSiteIsUpgradeable() {
+		if ( this.props.selectedSite && ! this.props.isSiteUpgradeable ) {
 			page.redirect( '/domains/add' );
 		}
-	},
+	}
 
-	handleDomainsAvailabilityChange: function( isAvailable ) {
+	handleDomainsAvailabilityChange( isAvailable ) {
 		this.setState( { domainRegistrationAvailable: isAvailable } );
-	},
+	}
 
-	handleAddRemoveDomain: function( suggestion ) {
+	handleAddRemoveDomain( suggestion ) {
 		if ( ! cartItems.hasDomainInCart( this.props.cart, suggestion.domain_name ) ) {
 			this.addDomain( suggestion );
 		} else {
 			this.removeDomain( suggestion );
 		}
-	},
+	}
 
 	handleAddMapping( suggestion ) {
 		upgradesActions.addItem( cartItems.domainMapping( { domain: suggestion.domain_name } ) );
-		page( '/checkout/' + this.props.sites.getSelectedSite().slug );
-	},
+		page( '/checkout/' + this.props.selectedSite.slug );
```

Comment:
Does it make sense to alter this to use the `getSelectedSiteSlug` selector for this one, so it'll be slightly more resistant to possible future structural changes to the `site` object?